modules/elastic/src/main/scala/ch/epfl/bluebrain/nexus/kg/query/instances/InstancesElasticQueries.scala

```diff
@@ -30,16 +33,17 @@ class InstancesElasticQueries[F[_]](elasticClient: ElasticClient[F], settings: E
     * @param schemaName   schema name
     * @param deprecated   boolean to decide whether to filter deprecated objects
     * @param published    boolean to decide whether to filter published objects
+    * @param acls         list of access controls to restrict the query
     * @return query results
     *
     */
   def list(pagination: Pagination,
            schemaName: SchemaName,
            deprecated: Option[Boolean],
-           published: Option[Boolean]): F[QueryResults[InstanceId]] = {
-    elasticClient.search[InstanceId](query(termsFrom(deprecated, published) :+ schemaGroupTerm(schemaName): _*))(
-      pagination,
-      sort = defaultSort)
+           published: Option[Boolean],
+           acls: FullAccessControlList): F[QueryResults[InstanceId]] = {
+    elasticClient.search[InstanceId](query(acls, termsFrom(deprecated, published) :+ schemaGroupTerm(schemaName): _*),
```

Comment:
query is not needed here. Just check if `acls` on the path `schemaName.domainId` (with the prefix) have read permissions and execute the query for that specific index. Otherwise just don't execute it. Similar to [this method](https://github.com/BlueBrain/nexus-kg/blob/v0/modules/sparql/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/query/builder/FilterQueries.scala#L48)

Reply:
we still need query for deprecated, published as well as for sorting, but I removed some unnecessary terms in some places.
There's still some room for optimisation though.